src/mono/System.Private.CoreLib/src/System/Environment.iOS.cs

```diff
@@ -34,20 +34,22 @@ private static string GetFolderPathCore(SpecialFolder folder, SpecialFolderOptio
 
         private static string? GetSpecialFolder(SpecialFolder folder)
         {
+#if TARGET_TVOS
+        string? DocumentsSearchPath = CombineAndCreateSearchPath(NSSearchPathDirectory.NSLibraryDirectory, Path.Combine("Caches", "Documents"));
+#else
+        string? DocumentsSearchPath = Interop.Sys.SearchPath(NSSearchPathDirectory.NSDocumentDirectory);
```

Comment:
this means we're fetching `DocumentsSearchPath` every time even for unrelated cases.
I'd rather refactor this so that we have something like this:

```csharp
static string CombineDocumentsDirectory(string subdirectory)
{
#if TARGET_TVOS
        string? path = CombineSearchPath(NSSearchPathDirectory. NSLibraryDirectory, "Caches", "Documents", subdirectory);
        // ... comment about tvOS special case....
        if (!Directory.Exists (path))
                    Directory.CreateDirectory (path);
#else
        string? path = CombineSearchPath(NSSearchPathDirectory.NSDocumentDirectory, subdirectory);
#endif

       return path;
}
```

and then we can use that where we use `NSSearchPathDirectory.NSDocumentDirectory` today (for `LocalApplicationData` we'd just do `CombineDocumentsDirectory(string.Empty)`)